scanner/cmd/scanner/main.go

```diff
@@ -221,7 +227,50 @@ func getServices(b *Backends) []grpc.APIService {
 	return srvs
 }
 
-// Close closes all the backends used by the scanner.
+// HealthCheck returns true if all configured backends are healthy and ready.
+func (b *Backends) HealthCheck(ctx context.Context) bool {
+	var checkList []func(context.Context) error
+	if b.Indexer != nil {
+		checkList = append(checkList, b.Indexer.Ready)
+	}
+	if b.RemoteIndexer != nil {
+		checkList = append(checkList, b.RemoteIndexer.Ready)
+	}
+	if b.Matcher != nil {
+		checkList = append(checkList, b.Matcher.Ready)
+	}
+	for _, check := range checkList {
+		if err := check(ctx); err != nil {
+			zlog.Error(ctx).Err(err).Msg("scanner is not ready")
+			return false
+		}
+	}
+	return true
+}
+
+// HealthRoutes returns HTTP routes for health checking the configured backends.
+func (b *Backends) HealthRoutes() (r []routes.CustomRoute) {
+	for n, h := range map[string]http.HandlerFunc{
+		"/health/liveness": func(w http.ResponseWriter, r *http.Request) {
+			// If we can reply, we are alive.
+			w.WriteHeader(http.StatusOK)
+		},
+		"/health/readiness": func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I just found out `startupProbe` exists, and that might be closer to what we want. Though, I think `readinessProbe` still effectively captures the same idea

Reply:
Startup prevents readiness and liveness from kicking in. What we are proposing here is readiness to remove traffic from the POD if it cannot handle traffic (DB's out, etc), and liveness to kill the POD if unresponsive (although you might argue that the proposed liveness is shallow).